packages/core/src/types.ts

```diff
@@ -697,6 +776,383 @@ export interface SettingsConfig {
 }
 /* eslint-enable @typescript-eslint/consistent-indexed-object-style -- needed to allow extension */
 
+/**
+ * The configuration for a set of files.
+ */
+export interface ConfigObject<Rules extends RulesConfig = RulesConfig> {
```

Comment:
Is it intended that the `basePath` property is not included, like in `Linter.Config`?

Reply:
Oops no. I had started working on this before `basePath` was added. I'll add it.